benchadapt/python/benchadapt/result.py

```diff
@@ -210,6 +214,48 @@ def to_publishable_dict(self):
         return res_dict
 
 
+def validate_or_remove_github_commit_key(res_dict: Dict, strict=False):
+    """
+    Mutate BenchmarkResult dictionary in-place to make its `github` key
+    property be compliant with the Conbench HTTP API.
```

Comment:
Maybe this sentence sets a higher goal than the code below fulfills. But then we know that the code below is buggy.